src/globalComponents/tankAPIIntegration.js

```diff
@@ -16,8 +19,52 @@ function getFavoriteTankID() : Promise<Response> {
 			'x-auth-token': getLoginToken()
 		},
 	});
-	return responsePromise;
+	responsePromise.then (
+		response => response.json().then(data => {
+			if (response.status !== 200) {
+				console.log(response.status);
+				console.log(data.msg);
+			}
+			else {
+				const tank = new BackendTank(
+					data._id,
+					data.components,
+					data.casusCode,
+					data.isBot,
+					data.userid,
+					data.tankName
+				);
+				onLoad(getTank(tank));
+			}
+		})
+	)
 }
+
+function setFavoriteTankId(tankId: string, onLoad:(response: boolean) => void) {
```

Comment:
Oh last thing. Make sure you put return types on all of these functions (void). So it should look like:
```function setFavoriteTankId(tankId: string, onLoad:(response: boolean) => void): void {```